File: src/difficulty_1/difficulty_1_2/fizzBuzz/FizzBuzz.java
package difficulty_1.difficulty_1_2.fizzBuzz;

import java.util.Scanner;

public class FizzBuzz {


    public static void main (String[] args) {
        Scanner scanner = new Scanner(System.in);
        int fizz = scanner.nextInt();
        int buzz = scanner.nextInt();

        int n = scanner.nextInt();

        int prod;
        if (buzz % fizz == 0) {
            prod = buzz;
        } else {
            prod = fizz * buzz;
        }

        for(int i = 1; i <= n; i++) {
            if(i % prod == 0 ) {
                System.out.println("FizzBuzz");
            } else if(i % fizz == 0) {
                System.out.println("Fizz");
            } else if(i % buzz == 0) {
                System.out.println("Buzz");
            } else {
                System.out.println(i);
            }
        }
    }
}

File: KotlinImplementation/src/difficulty_1/FizzBuzz.kt
package difficulty_1

fun main() {
    val (X, Y, N) = readLine()!!.split(' ').map(String::toInt)

    for(i in 1..N) {
        if(i % X == 0 && i % Y == 0) {
            println("FizzBuzz")
            continue
        }
        if(i % X == 0) {
            println("Fizz")
            continue
        }
        if(i % Y == 0) {
            println("Buzz")
            continue
        }
        println(i)
    }
}